### ISSUE TITLE
Code is not up to date with launchpad

### ISSUE THREAD

**Issue by WebDrake** [ubports/unity-system-compositor#3]
(Context: I've been checking out the UBPorts repo to see if I can move my MirAL refactoring work over here.)

It looks like the ubports code has some patches missing compared to the original launchpad upstream.

The `xenial` branch is missing a lot of commits (I'm not sure why).  The `bionic` branch appears to be up to date apart from the most recent merge commit in https://code.launchpad.net/~unity-system-compositor-team/unity-system-compositor/trunk (and associated subsidiary patches).

The missing patches are probably no big deal in the grand scheme of things, but it'd be nice to have everything up to date with the most recent state of the original USC project.

On a related note: I don't know what UBPorts' branch/code management policy is, but given that UBPorts is going to be upstream for USC now, it'd be nice to have development branches separate from distro-specific packaging.  Ideally I'd suggest a semver-based approach (i.e. separate branches for major and minor version releases), similar to how things were traditionally managed on Launchpad IIRC (see: https://launchpad.net/unity-system-compositor/+series).

If possible, I'd like to get this dealt with before migrating my existing refactoring work over here.

**Comment by mariogrip** [comment]
The xenial branch is still older due to mobile, on mobile we still use mir 0.24 at this first release of xenial. The current stable mobile release is still on vivid, that's why we have decided to first do the base upgrade before starting upgrading mir components to be able to have a stable xenial base in a short time as possible. Since our CI system is directly tied up to the branches, why our branches are like they are. But once mobile catch up with desktop we are looking to have one "master" branch.

But for now "bionic" is "master"

**Comment by mariogrip** [comment]
Later on once we get to a more "stable" and "usable" state for desktop, we have different release channels, but right now, the latest development build will be automatically pushed to the repos.

**Comment by WebDrake** [comment]
OK, makes sense.  The patches from the `bionic` branch might be possible to ignore: they are all updates to the `debian/rules`.  You can see them in the `launchpad` branch on my GitHub fork:
https://github.com/WebDrake/unity-system-compositor-yunit/commits/launchpad (it's all the June 2017 commits from Alan Griffiths).

Do these look relevant to you?  If not then let's close this issue and I'll submit a PR against the `bionic` branch.

**Comment by mariogrip** [comment]
They don't look that relevant to me, so we can ignore those. 

Ok, will close this.
